packages/protocol-rest/src/RestSerializer.ts

```diff
@@ -244,4 +249,34 @@ export class RestSerializer<StreamType> implements
             }
         }
     }
+
+    /**
+     * Add Content-Type header for rest-json protocol explicitly
+     * If payload is supplied in input, the content-type should be set according to payload shape;
+     * If payload is specified but not supplied in input, no content-type header is needed;
+     * If there's no payload in input shape, set content-type as 'application/json';
+     * @param operation 
+     */
+    private populateContentTypeHeader(operation: OperationModel, input: any): HeaderBag {
+        const contentTypeHeader = {};
+        const {
+            input: inputShape,
+            metadata: {protocol}
+        } = operation;
+        if (protocol !== 'rest-json') return contentTypeHeader
+        if (typeof (inputShape.shape as Structure).payload === 'string') {
+            const payloadMemberName = (inputShape.shape as Structure).payload!;
```

Comment:
What is the `!` doing at the end of this line?

Reply:
The `payload` can be string or undefined. But in fact the payload here cannot be undefined because of the if statement. The `!` is to persuade Typescript that `payload` always exists